### ISSUE TITLE
Fix Lint/ParenthesesAsGroupedExpression false positive with yield

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop/rubocop#15168]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/ParenthesesAsGroupedExpression

### Steps to reproduce the problem

1. Create a `example.rb` file.
2. Add this line: `def rewrite = read.then { |content| write yield(content) if block_given? }`
3. Run `rubocop example.rb`.

### Expected behavior

I would not expect this to cause an error. This was working fine prior to RuboCop 1.86.2.

### Actual behavior

Here's what you see first:

``` 
[Correctable] Lint/ParenthesesAsGroupedExpression: yield(content) interpreted as grouped expression.
```

When running auto-correct, this gets worse, as you end up with this code which is syntactically incorrect:

``` ruby
def rewrite = read.then { |content| writeyield(content) if block_given? }
```

Notice that the `write` and `yield` message are completely collapsed as a `writeyield` message.

### RuboCop configuration

Here's my full [configuration](https://github.com/bkuhlmann/caliber/blob/main/config/all.yml) if it helps.

### RuboCop version

```shell
1.86.2 (using Parser 3.3.11.1, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 4.0, running on ruby 4.0.4) [arm64-darwin25.4.0]
  - rubocop-capybara 2.23.0
  - rubocop-disable_syntax 0.2.0
  - rubocop-packaging 0.6.0
  - rubocop-performance 1.26.1
  - rubocop-thread_safety 0.7.3
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
```

**Comment by bkuhlmann** [comment]
:information_source: I should note that using extra parentheses fixes the issue but is heavy handed. Example:

``` 
# This works.
def rewrite = read.then { |content| write(yield(content)) if block_given? }
```
